tests/e2e-test/internal/runner/util.go

```diff
@@ -42,7 +42,7 @@ func gitCommit(d, msg string) error {
 	if err != nil {
 		return fmt.Errorf("git config error: %w, output: %s", err, o)
 	}
-	o, err = runCommand(d, "git", []string{"commit", "-m", msg})
+	o, err = runCommand(d, "git", []string{"commit", "-m", msg, "--allow-empty"})
```

Comment:
In what scenario, there will be no diff?

Reply:
I am not familiar with this function. But if a function is used to validate the resources then there might be no diff.

Reply:
When the hydration pipeline work is done, ideally, this function can be specified in the validation section and then the test harness doesn't need to do `--allow-empty`.
We can improve that later.

Reply:
Another possibility is the transformer is conditional.